### PR TITLE
Load test recipes with cluster loader 2.

### PR DESCRIPTION
**Pull request by SHRUTI6991**
This PR creates 4 custom test recipes that performs 4 different tests. These test can be used to run different scenarios with different cloud providers.

1. `throughput-test` - This test allows varying the creation rate QPS and checking how QPS variation affects Pod scheduling and ready rate.
2. `medium-scale-concurrent-load-test` - This test allows the simultaneous creation and deletion of sandboxes to determine the latency during varying sustained loads.
3. `high-volume-test` - This test linearly scales up sandboxes at a constant QPS to check how increasing the sandboxes affect agent controller and the cluster.
4. `warmpool-burst-test` - This test calculates the pod acquisition latency for cold start. Currently we don't emit any warm pool acquisition metrics.

### CODE REVIEW

File: dev/load-test/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+To run the test against your Kubernetes cluster, execute the command below:
+
+```bash
+./clusterloader2 
+--testconfig=../../agent-sandbox/dev/load-test/agent-sandbox-load-test.yaml 
+--kubeconfig=$HOME/.kube/config
+--provider=gke
+```
+
+To run the test against against your local kind Kubernetes cluster, please follow
```
Comment (trainyao) [inline dev/load-test/README.md]: "against against" typo

File: dev/load-test/test-recipes/warmpool-test/warmpool-burst-test.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    - basename: warmpool
+      objectTemplatePath: "cluster-loader-warmpool.yaml"
+      templateFillMap:
+        TemplateName: perf-template-warm-0
+        Replicas: 100
+    waits:
+    - type: Running
+      labelSelector: "agents.x-k8s.io/pool"
+
+# --- 2. DRAIN WARMPOOL (Excluded from Cold Measurement) ---
```
Comment (vicentefb) [inline dev/load-test/test-recipes/warmpool-test/warmpool-burst-test.yaml:51]: I understand the intent is to establish a baseline for Cold Start latency after exhausting the warm pool. But just curious shouldn't we also measure the drain phase (Step 2)? Because afaiu, the test tells us 'How bad is it when the pool is empty?' (Cold Start). But it doesn't tell us 'How good is it when the pool is full?' (Warm Acquisition).

File: dev/load-test/cluster-loader-sandbox-template.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  name: {{.Name}}
+spec:
+  podTemplate:
+    metadata:
+      labels:
+        latency-type: {{.LatencyType}}
+    spec:
+      restartPolicy: Never 
+      runtimeClassName: gvisor
```
Comment (vicentefb) [inline dev/load-test/cluster-loader-sandbox-template.yaml]: Can we template this to `{{.RuntimeClass}}` ? This would allow us to run these load tests against local clusters (e.g., Kind with runc) or accept other runtimeClassNames in general. It could also help for CI verification as well.

**Review by vicentefb — APPROVED** [review]
lgtm

File: dev/load-test/test-recipes/warmpool-test/warmpool-burst-test.yaml (hunk 2 above)
Comment (SHRUTI6991) [reply dev/load-test/test-recipes/warmpool-test/warmpool-burst-test.yaml:51]: Yes, currently we don't have a Prometheus metric that calculates warm acquistion. The current best calculation is the P100 time we get from "Acquire from Warmpool" step since all 100 pods are acquired in 1 step for 100 replicas. Let me add a todo here to calc this.

**Comment by SHRUTI6991** [comment]
> Please squash /fix the commits. We shouldn't have any "merge" commits in the PR; we use a rebase flow.
> 
> Also all your commits have a big UNVERIFIED flag next to them... is that something you can fix?

I have squashed all the commits into 1.
